lib/viewcomponents/gui/toolbar.jsx

```diff
@@ -0,0 +1,94 @@
+var React = require('react');
+var classSet = React.addons.classSet;
+
+var ObjectUtils = require('../../utilities/objects');
+
+var Panel = require('./panel.jsx');
+/*
+tools: an object of tool config objects keyed by name {
+  panelContent: constructed component of contents of the panel,
+  icon: path to icon image
+  title: string title of panel
+  props: any props to pass to the panel
+}
+ */
+
+
+var Toolbar = React.createClass({
+  propTypes: {
+    tools: React.PropTypes.object
+  },
+
+  getDefaultProps: function() {
+    return {
+      tools: {}
+    };
+  },
+
+  getInitialState: function() {
+    return {
+      activePanel: null
+    };
+  },
+
+  render: function() {
+    var createButtons = function() {
```

Comment:
On the other hand, that would make the render method very short and it wouldn't describe the contents very explicitly. Hm.